src/conf_mode/containers.py

```diff
@@ -206,14 +207,35 @@ def apply(container):
                       tmp += ' --ipv6'
                 _cmd(tmp)
 
+
+        # disable masquerading and use traditional bridging so VyOS can control firewalling/NAT/etc on the cni-podmanX interface
+        for network, network_config in container['network'].items():
+            if 'enable_bridging' in network_config:
+                # podman doesn't expose these options
+                # this is the easiest way to change properties in the network config
+                with open('/etc/cni/net.d/%s.conflist' % network, 'r') as sources:
+                    lines = sources.readlines()
+                with open('/etc/cni/net.d/%s.conflist' % network, 'w') as sources:
+                    for line in lines:
+                        line = re.sub(r'"ipMasq": true', '"ipMasq": false', line)
+                        sources.write(line)
+
     # Add container
     if 'name' in container:
         for name, container_config in container['name'].items():
             # Check if the container has already been created
             if not container_exists(name):
                 image = container_config['image']
-                # Currently the best way to run a command and immediately print stdout
-                print(os.system(f'podman pull {image}'))
+
+                if "containerfile" in container_config:
```

Comment:
This either "builds" the container from the Dockerfile, or pulls it from a registry. A `verify()` step should be added so one can either specify `image` or `containerfile` but not both at the same time as the `containerfile` will win.

Reply:
These aren't exclusionary.  The image name is still used for local resolution, and in fact, required to avoid auto-generated names